src/recallcard/randomcard.php

```diff
@@ -22,17 +22,31 @@
 }
 if ($type == "lesson") {
 	$lesson = get_lesson($getid);
-	$group = $conn->query("SELECT * FROM `group` WHERE `id` = '{$data_lesson['group']}'");
+	$stmt = $conn->prepare("SELECT * FROM `group` WHERE `id` = ?");
+	$stmt->bind_param("i", $data_lesson['group']);
+	$stmt->execute();
+	$thisgroup = $stmt->get_result();
+	$data_thisgroup = $thisgroup->fetch_array();

```

Comment:
There is a duplicate `->fetch_array()` (this line and the following line)

Reply:
You can replace this line and the line above with `$group = $stmt->get_result();` and the following line will continue doing its job

Reply:
Done.

Reply:
It seems that `$data_group` variable disappears